codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerBuilderGeneratorTest.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy.generators
+
+import org.junit.jupiter.api.Test
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.rust.codegen.core.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.core.rustlang.rust
+import software.amazon.smithy.rust.codegen.core.smithy.generators.StructureGenerator
+import software.amazon.smithy.rust.codegen.core.smithy.generators.implBlock
+import software.amazon.smithy.rust.codegen.core.testutil.asSmithyModel
+import software.amazon.smithy.rust.codegen.core.testutil.compileAndTest
+import software.amazon.smithy.rust.codegen.core.testutil.generatePluginContext
+import software.amazon.smithy.rust.codegen.core.util.lookup
+import software.amazon.smithy.rust.codegen.server.smithy.renderDefaultBuilder
+import software.amazon.smithy.rust.codegen.server.smithy.testutil.serverTestCodegenContext
+import software.amazon.smithy.rust.codegen.server.smithy.testutil.serverTestSymbolProvider
+import kotlin.io.path.createDirectory
+import kotlin.io.path.writeText
+
+class ServerBuilderGeneratorTest {
+    @Test
+    fun `generate default values`() {
+        val model =
+            """
+            namespace com.test
+            use aws.protocols#restJson1
+            use smithy.framework#ValidationException
+
+            structure MyStruct {
```

Comment:
We need _a lot_ more testing.

We need to test all shape combinations (including documents and collections), with and without `@required`, with `@default(null)`  etc.

_Ideally_, we should also test binding of the members to all parts of the HTTP message (headers, query string, uri path, payload etc). But that's not a strict requirement, since it shouldn't affect our builder logic and this is a unit test after all. Still, it should be easy to do.

I suggest you synthesize the model programmatically so that you don't have to manually write out each test case.

The tests should assert both cases when the user sets a value and the default does not get enacted, and when they don't and the default does get enacted.